examples/graphql-file-upload/graphql_file_upload.bal

```diff
@@ -0,0 +1,56 @@
+import ballerina/graphql;
+import ballerina/io;
+
+service /fileUpload on new graphql:Listener(4000) {
+
+    // Remote functions can use the `graphql:Upload` type as an input
+    // parameter type.
+    remote function singleFileUpload(graphql:Upload file)
+        returns string|error {
+
+        // Access the file name from the `graphql:Upload` type parameter.
+        // Similarly, it can access the mime type as `file.mimeType`
+        // and encoding as `file.encoding`. Except the `byteStream` field, all
+        // other fields in the `graphql:Upload` are `string` values.
+        string fileName = file.fileName;
+        string path = string`./uploads/${fileName}`;
+
+        // Access the byte stream of the file from the `graphql:Upload` type
+        // parameter. The type of the `byteStream` field is
+        // `stream<byte[], io:Error?>`
+        stream<byte[], io:Error?> byteStream = file.byteStream;
+
+        // Store the received file using the ballerina `io` package. If any
+        // `error` occurred during the file write, it can be returned as the
+        // resolver function output.
+        check io:fileWriteBlocksFromStream(path, byteStream);
+
+        // Returns the message if the uploading process is successful.
+        return "Successfully Uploaded";
+    }
+
+    string[] uploadedFiles = [];
```

Comment:
Let's define this at the top of the service. Better to mention the reason for creating this as well.

Reply:
Can't we have it inside the remote function itself ?

Reply:
This is to add the resource function. This array shares information between the remote and resource functions. 

There are two options here:
1. Remove this `uploadedFiles` array, and add another simple resource function. Maybe a simple `greeting` resource.
2. Explain why we have this.